Create DataBase

Spelaya-melon-X opened the pull request.

Spelaya-melon-X committed:
I am doing some functions and write some sql requests (if build dir )

Spelaya-melon-X committed:
i am fixed the code operation

Spelaya-melon-X committed:
add foto +normal  READMDE

Spelaya-melon-X committed:
Update README.md

Spelaya-melon-X committed:
Update README.md

Spelaya-melon-X committed:
Update README.md

Spelaya-melon-X committed:
Update README.md

Spelaya-melon-X committed:
I have created my own classes and methods for each of the tables in the database

Spelaya-melon-X committed:
I got rid of unnecessary files.

Spelaya-melon-X committed:
add fucntion for read vector all student_id and person_id

Vdaleke commented on CMakeLists.txt:
Комментарии на русском языке, заменить на английский

Vdaleke commented on CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Название проекта (можете изменить)
+project(CreateAllTables)
```
Когда отребейзишься на новый main, заменить

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+    program_id SERIAL PRIMARY KEY,
+    program_name VARCHAR(50) UNIQUE NOT NULL
+);
+INSERT INTO Programs (program_name) VALUES 
```
Это не то должно быть заинлайнено, а должно настраиваться.

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+    role_id SERIAL PRIMARY KEY,
+    role_name VARCHAR(255) UNIQUE NOT NULL
+);
+INSERT INTO Roles (role_name) VALUES 
```
оставить TODO, что потом добавим разные роли.

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+
+CREATE TABLE Students (
+    student_id SERIAL PRIMARY KEY,
+    person_id INT UNIQUE REFERENCES People(person_id) ON DELETE CASCADE,
```
почему нет NOT NULL?

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+);
+
+CREATE TABLE Person_Role (
+    person_id INT REFERENCES People(person_id) ON DELETE CASCADE,
```
почему нет NOT NULL?

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+    subject_name VARCHAR(255) UNIQUE NOT NULL
+);
+
+INSERT INTO Subjects (subject_name)
```
должно быть настраиваемо не в коде

Vdaleke commented on CreateAllTable.sql:
```diff
@@ -0,0 +1,139 @@
+
+
+CREATE TABLE Teacher_Subject (
+    teacher_id INT REFERENCES People(person_id) ON DELETE CASCADE,
```
not null. как и везде дальше где нужно

Vdaleke commented on README.md:
```diff
@@ -1,2 +1,4 @@
 # EntryPoint
-Проект на плюсах от 3 амбициозных пацанчиков 
+
+database structure:
```
переместить в docs/

Vdaleke commented on build/.DS_Store:
должно быть в .gitignore

Vdaleke commented on build/CMakeCache.txt:
должно быть в .gitignore

Vdaleke commented on build/CMakeFiles/3.31.5/CMakeCCompiler.cmake:
папка build целиком должна быть в .gitignore

Vdaleke commented on inc/Categories.hpp:
будет include папка, inc никто не пишет

Vdaleke commented on inc/Categories.hpp:
```diff
@@ -0,0 +1,23 @@
+
+class Categories {
+public:
+    static void Create(pqxx::connection &conn,  const std::string category_name , const int filter_id , bool required); 
```
тут явно не проходился clang-format. пройтись.

Vdaleke commented on inc/CreateAllTables.hpp:
под такое отдельный хедер сомнительный. вынести все такие функции в какие-нибудь utils.cpp или tools.cpp

Vdaleke commented on inc/Feedback.hpp:
```diff
@@ -0,0 +1,17 @@
+public:
+    static void Create(pqxx::connection &conn, const int teacher_id , const int subject_id , std::string feedback_name); 
+
+    static std::string  ReadFeedbackName(pqxx::connection &conn, const int feedback_id); 
```
Не надо делать различные read функции под каждое поле. сделай чтобы возвращался целиком класс

Vdaleke commented on inc/Feedback_Participants.hpp:
```diff
@@ -0,0 +1,15 @@
+public:
+    static void Create(pqxx::connection &conn,  int student_id , int feedback_id , std::string student_feedback); 
+
+    static std::string  ReadFeedback(pqxx::connection &conn, const int student_id ,const int feedback_id );   
```
Аналогично целиком возвращать класс. и так везде.

Vdaleke commented on src/Categories.cpp:
Ты задолбаешься std::cout чистить. Добавьте нормальную библиотеку для логирования. Например https://github.com/gabime/spdlog

Vdaleke reviewed (changes requested):
Я что-то накидал, что быстро увидел. Но это не полное ревью

Vdaleke commented:
Сделать нормальное название пул реквеста, написать нормальное description. Почистить историю коммитов. Сделать логическое разделение коммитов по смыслу.

Spelaya-melon-X committed:
FOR MATVEY : examples of code templates that I want to rewrite the rest of the code into

Spelaya-melon-X committed:
FOR MATVEY : A simple test has been written , and the functions via the template have been improved .

Spelaya-melon-X committed:
FOR MATVEY : edit bags

Spelaya-melon-X committed:
in proccess...

Spelaya-melon-X committed:
Create All Tables

Spelaya-melon-X committed:
create 1 part table

Spelaya-melon-X committed:
eddit database photo

EgorLukavenko committed:
feat(db): add Feedback_URL and Feedback_Request tables; refactor Filters, Tags, and Categories
- Added new tables: Feedback_URL and Feedback_Request
- Refactored existing tables: Filters, Tags, and Categories

Spelaya-melon-X committed:
corrections Bad Tables

Spelaya-melon-X committed:
CORRECTING classes , i am do normal Read Function

Spelaya-melon-X committed:
Add classes for Create All tables and Delete all tables

Spelaya-melon-X committed:
rewrite comments from Russian to English and delete unnecessary files

Spelaya-melon-X committed:
merge dev/kalashnikov and pre-main

Spelaya-melon-X committed:
delete inc src

Spelaya-melon-X committed:
delete trash file

Spelaya-melon-X committed:
inc --> inclue in .hpp file

Spelaya-melon-X committed:
The implementation of BaseCrud has been changed, a new class has been added to take all students and people from tables, and the Update and Delete methods for category tables have been corrected.

Spelaya-melon-X committed:
do all

Spelaya-melon-X committed:
do all

Spelaya-melon-X committed:
add new function + upate all

Spelaya-melon-X committed:
add new

Spelaya-melon-X committed:
bla bla

Spelaya-melon-X committed:
bla blaa

Spelaya-melon-X committed:
bla blaaaa

Spelaya-melon-X committed:
add TODO list for comand

Spelaya-melon-X committed:
delete old version and create todo list for comadmd

Spelaya-melon-X committed:
debug problems , add new function for my teem

Spelaya-melon-X committed:
erroe debug

Spelaya-melon-X committed:
f

Spelaya-melon-X committed:
fa

Spelaya-melon-X committed:
debug

Spelaya-melon-X committed:
nafig int

Spelaya-melon-X committed:
DeletePerson now with tg_nick

Spelaya-melon-X committed:
DeletePerson now with tg_nick

Spelaya-melon-X committed:
debug delete functuons

Spelaya-melon-X committed:
bla

Spelaya-melon-X committed:
bla

Spelaya-melon-X committed:
dop_function

Spelaya-melon-X committed:
bla
